File: backend/internal/service/sales_service.go
package service

import (
	"context"

	"github.com/water-factory/api/internal/domain"
	"github.com/water-factory/api/internal/dto"
	"github.com/water-factory/api/internal/repository"
)

type SalesService struct {
	repo *repository.SalesRepo
}

func NewSalesService(repo *repository.SalesRepo) *SalesService {
	return &SalesService{repo: repo}
}

// Vehicles
func (s *SalesService) ListVehicles(ctx context.Context) ([]*domain.Vehicle, error) {
	return s.repo.ListVehicles(ctx)
}

// Sales Orders
func (s *SalesService) ListSalesOrders(ctx context.Context) ([]*domain.SalesOrder, error) {
	return s.repo.ListSalesOrders(ctx)
}
func (s *SalesService) GetSalesOrder(ctx context.Context, id string) (*domain.SalesOrder, error) {
	return s.repo.GetSalesOrder(ctx, id)
}
func (s *SalesService) CreateSalesOrder(ctx context.Context, userID string, req *dto.CreateSalesOrderRequest) (*domain.SalesOrder, error) {
	return s.repo.CreateSalesOrder(ctx, userID, req)
}
func (s *SalesService) UpdateSalesOrder(ctx context.Context, id string, req *dto.UpdateSalesOrderRequest) (*domain.SalesOrder, error) {
	return s.repo.UpdateSalesOrder(ctx, id, req)
}
func (s *SalesService) ConfirmSalesOrder(ctx context.Context, id string) (*domain.SalesOrder, error) {
	return s.repo.ConfirmSalesOrder(ctx, id)
}
func (s *SalesService) CancelSalesOrder(ctx context.Context, id string) (*domain.SalesOrder, error) {
	return s.repo.CancelSalesOrder(ctx, id)
}

// Delivery Orders
func (s *SalesService) ListDeliveryOrders(ctx context.Context) ([]*domain.DeliveryOrder, error) {
	return s.repo.ListDeliveryOrders(ctx)
}
func (s *SalesService) GetDeliveryOrder(ctx context.Context, id string) (*domain.DeliveryOrder, error) {
	return s.repo.GetDeliveryOrder(ctx, id)
}
func (s *SalesService) CreateDeliveryOrder(ctx context.Context, req *dto.CreateDeliveryOrderRequest) (*domain.DeliveryOrder, error) {
	return s.repo.CreateDeliveryOrder(ctx, req)
}
func (s *SalesService) DispatchDeliveryOrder(ctx context.Context, id string, userID string) (*domain.DeliveryOrder, error) {
	return s.repo.DispatchDeliveryOrder(ctx, id, userID)
}
func (s *SalesService) DeliverDeliveryOrder(ctx context.Context, id string) (*domain.DeliveryOrder, error) {
	return s.repo.DeliverDeliveryOrder(ctx, id)
}

// Invoices
func (s *SalesService) ListInvoices(ctx context.Context) ([]*domain.Invoice, error) {
	return s.repo.ListInvoices(ctx)
}
func (s *SalesService) GetInvoice(ctx context.Context, id string) (*domain.Invoice, error) {
	return s.repo.GetInvoice(ctx, id)
}
func (s *SalesService) CreateInvoice(ctx context.Context, userID string, req *dto.CreateInvoiceRequest) (*domain.Invoice, error) {
	return s.repo.CreateInvoice(ctx, userID, req)
}
func (s *SalesService) MarkInvoicePaid(ctx context.Context, id string, req *dto.MarkPaidRequest) (*domain.Invoice, error) {
	return s.repo.MarkInvoicePaid(ctx, id, req)
}

File: frontend/src/app/core/services/sales.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import {
  SalesOrder, DeliveryOrder, Invoice, Vehicle,
  CreateSalesOrderRequest, CreateDeliveryOrderRequest,
  CreateInvoiceRequest, MarkPaidRequest,
} from '../models/sales.model';

@Injectable({ providedIn: 'root' })
export class SalesService {
  private http = inject(HttpClient);
  private base = '/api/v1/sales';

  // Vehicles
  listVehicles(): Observable<Vehicle[]> {
    return this.http.get<{ data: Vehicle[] }>(`${this.base}/vehicles`).pipe(map(r => r.data));
  }

  // Sales Orders
  listSalesOrders(): Observable<SalesOrder[]> {
    return this.http.get<{ data: SalesOrder[] }>(`${this.base}/orders`).pipe(map(r => r.data));
  }
  getSalesOrder(id: string): Observable<SalesOrder> {
    return this.http.get<{ data: SalesOrder }>(`${this.base}/orders/${id}`).pipe(map(r => r.data));
  }
  createSalesOrder(req: CreateSalesOrderRequest): Observable<SalesOrder> {
    return this.http.post<{ data: SalesOrder }>(`${this.base}/orders`, req).pipe(map(r => r.data));
  }
  updateSalesOrder(id: string, req: Partial<CreateSalesOrderRequest>): Observable<SalesOrder> {
    return this.http.put<{ data: SalesOrder }>(`${this.base}/orders/${id}`, req).pipe(map(r => r.data));
  }
  confirmSalesOrder(id: string): Observable<SalesOrder> {
    return this.http.post<{ data: SalesOrder }>(`${this.base}/orders/${id}/confirm`, {}).pipe(map(r => r.data));
  }
  cancelSalesOrder(id: string): Observable<SalesOrder> {
    return this.http.post<{ data: SalesOrder }>(`${this.base}/orders/${id}/cancel`, {}).pipe(map(r => r.data));
  }

  // Delivery Orders
  listDeliveryOrders(): Observable<DeliveryOrder[]> {
    return this.http.get<{ data: DeliveryOrder[] }>(`${this.base}/delivery-orders`).pipe(map(r => r.data));
  }
  getDeliveryOrder(id: string): Observable<DeliveryOrder> {
    return this.http.get<{ data: DeliveryOrder }>(`${this.base}/delivery-orders/${id}`).pipe(map(r => r.data));
  }
  createDeliveryOrder(req: CreateDeliveryOrderRequest): Observable<DeliveryOrder> {
    return this.http.post<{ data: DeliveryOrder }>(`${this.base}/delivery-orders`, req).pipe(map(r => r.data));
  }
  dispatchDeliveryOrder(id: string): Observable<DeliveryOrder> {
    return this.http.post<{ data: DeliveryOrder }>(`${this.base}/delivery-orders/${id}/dispatch`, {}).pipe(map(r => r.data));
  }
  deliverDeliveryOrder(id: string): Observable<DeliveryOrder> {
    return this.http.post<{ data: DeliveryOrder }>(`${this.base}/delivery-orders/${id}/deliver`, {}).pipe(map(r => r.data));
  }

  // Invoices
  listInvoices(): Observable<Invoice[]> {
    return this.http.get<{ data: Invoice[] }>(`${this.base}/invoices`).pipe(map(r => r.data));
  }
  getInvoice(id: string): Observable<Invoice> {
    return this.http.get<{ data: Invoice }>(`${this.base}/invoices/${id}`).pipe(map(r => r.data));
  }
  createInvoice(req: CreateInvoiceRequest): Observable<Invoice> {
    return this.http.post<{ data: Invoice }>(`${this.base}/invoices`, req).pipe(map(r => r.data));
  }
  markInvoicePaid(id: string, req: MarkPaidRequest): Observable<Invoice> {
    return this.http.post<{ data: Invoice }>(`${this.base}/invoices/${id}/pay`, req).pipe(map(r => r.data));
  }
}
